rust/src/pgsql/pgsql.rs

```diff
@@ -0,0 +1,853 @@
+            0x61, 0x69, 0x6c, 0x73, 0x74, 0x6f, 0x72, 0x65, 0x00, 0x00,
+        ];
+        assert!(!probe_ts(buf));
+    }
+
+    #[test]
+    fn test_response_probe() {
+        /* Authentication Request MD5 password salt value f211a3ed */
+        let buf: &[u8] = &[
+            0x52, 0x00, 0x00, 0x00, 0x0c, 0x00, 0x00, 0x00, 0x05, 0xf2, 0x11, 0xa3, 0xed,
+        ];
+        assert!(probe_tc(buf));
+
+        /* R  8 -- Authentication Cleartext */
+        let buf: &[u8] = &[0x52, 0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0x03];
+        assert!(probe_tc(buf));
+
+        let buf: &[u8] = &[
+            /* R */ 0x52, /* 54 */ 0x00, 0x00, 0x00, 0x36, /* 12 */ 0x00, 0x00,
+            0x00, 0x0c, /* signature */ 0x76, 0x3d, 0x64, 0x31, 0x50, 0x58, 0x61, 0x38, 0x54,
+            0x4b, 0x46, 0x50, 0x5a, 0x72, 0x52, 0x33, 0x4d, 0x42, 0x52, 0x6a, 0x4c, 0x79, 0x33,
+            0x2b, 0x4a, 0x36, 0x79, 0x78, 0x72, 0x66, 0x77, 0x2f, 0x7a, 0x7a, 0x70, 0x38, 0x59,
+            0x54, 0x39, 0x65, 0x78, 0x56, 0x37, 0x73, 0x38, 0x3d,
+        ];
+        assert!(probe_tc(buf));
+
+        /* S   26 -- parameter status application_name psql*/
+        let buf: &[u8] = &[
+            0x53, 0x00, 0x00, 0x00, 0x1a, 0x61, 0x70, 0x70, 0x6c, 0x69, 0x63, 0x61, 0x74, 0x69,
+            0x6f, 0x6e, 0x5f, 0x6e, 0x61, 0x6d, 0x65, 0x00, 0x70, 0x73, 0x71, 0x6c, 0x00,
+        ];
+        assert!(probe_tc(buf));
+    }
+
+    #[test]
+    fn test_request_events() {
+        let mut state = PgsqlState::new();
+        // an SSL Request
+        let buf: &[u8] = &[0x00, 0x00, 0x00, 0x08, 0x04, 0xd2, 0x16, 0x2f];
+        state.parse_request(buf);
+        let ok_state = PgsqlStateProgress::SSLRequestReceived;
+
+        assert_eq!(state.state_progress, ok_state);
+
+        // TODO add test for startup request
+    }
+
+    #[test]
+    fn test_incomplete_request() {
+        let mut state = PgsqlState::new();
+        // An SSL Request
+        let buf: &[u8] = &[0x00, 0x00, 0x00, 0x08, 0x04, 0xd2, 0x16, 0x2f];
+
+        let r = state.parse_request(&buf[0..0]);
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: 0,
+                consumed: 0,
+                needed: 0
+            }
+        );
+
+        let r = state.parse_request(&buf[0..1]);
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: 1,
+                consumed: 0,
+                needed: 2
+            }
+        );
+
+        let r = state.parse_request(&buf[0..2]);
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: 1,
+                consumed: 0,
+                needed: 3
+            }
+        );
+
+        // This is the first message and only the first message.
+        let r = state.parse_request(buf);
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: 0,
+                consumed: 0,
+                needed: 0
+            }
```

Comment:
Does this mean that the request was complete and was fully processed? I'm a little confused because of the name of the test.

Reply:
It means, you're right, this one is out of place, here, given the test's name...